src/main/kotlin/lotto/domain/Money.kt

```diff
@@ -0,0 +1,51 @@
+package lotto.domain
+
+import java.math.BigDecimal
+import java.math.RoundingMode
+
+data class Money(val value: Long) : Comparable<Money> {
+
+    init {
+        check(value >= 0) { "돈은 음수가 될 수 없습니다. value: $value" }
```

Comment:
`check()`는 값을 만족하지 않으면 `IllegalStateException`을 발생시킵니다. `IllegalStateException` 대신 `IllegalArgumentException`을 사용하면 어떨까요? 아래 코멘트에 대해 고민해 봐요.
https://github.com/next-step/kotlin-lotto/pull/108#discussion_r592025667

```suggestion
        require(value >= 0) { "돈은 음수가 될 수 없습니다. value: $value" }
```

Reply:
@wotjd243 

지금 껏 `init{ }` 내부에서 프로퍼티 검증 시 `check`와 `require` 를 혼용하여 쓰곤 했는데요.

그만큼 두 예외 중 어떤 예외가 어울리는가에 대해서 명확하게 결정을 못했기 때문이에요.

check/require 둘 중 어느 것이 옳은가는 다음과 같은 관점으로 바꿔볼 수 있을 것 같아요.

`init { }`내부에서 value 는 **상태인가** / **인자인가**

&nbsp;

**상태라고 생각하는 이유**
`init{ }` 이 호출되는 시점은 해당 객체가 불완전한 시점.
따라서 `value` 는 불완전한 객체의 상태이다.

또한, Money 클래스를 자바로 디컴파일을 해보면 생성자 내부에서 `this.value(상태)`로 유효성검사를 하고 있다.
<img src="https://user-images.githubusercontent.com/25674959/112748864-aacb2200-8ff9-11eb-9e5e-e40e268700bb.png" width=400>

&nbsp;

**인자라고 생각하는 이유**
위에 `this.value` 로 검증을 하였지만, 결국에는 생성자 인자로 들어온 값이니 생성자의 인자를 검증한 것이라고 봐야한다.

위 두가지의 생각이 머리속에 공존하여 하나를 결정하지 못하는 듯 합니다. 지혜를 나눠주세요 :sos: